addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/handler/YamahaZoneThingHandler.java

```diff
@@ -92,20 +99,26 @@ public YamahaZoneThingHandler(Thing thing) {
     /**
      * Sets the {@link DeviceInformationState} for the handler.
      */
-    public void setDeviceInformationState(DeviceInformationState deviceInformationState) {
-        this.deviceInformationState = deviceInformationState;
+    public DeviceInformationState getDeviceInformationState() {
+        return getBridgeHandler().getDeviceInformationState();
     }
 
     @Override
     public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
         validateConfigurationParameters(configurationParameters);
 
+        logger.trace("Updating configuration of {}", getThing().getLabel());
+
         Configuration configuration = editConfiguration();
-        for (Entry<String, Object> configurationParmeter : configurationParameters.entrySet()) {
-            configuration.put(configurationParmeter.getKey(), configurationParmeter.getValue());
+        for (Entry<String, Object> configurationParameter : configurationParameters.entrySet()) {
+            configuration.put(configurationParameter.getKey(), configurationParameter.getValue());
         }
 
         updateConfiguration(configuration);
+
+        YamahaConfigurationUtil.<BigDecimal>readSettingOptional(thing, CONFIG_RELVOLUMECHANGE, v -> zoneConfiguration.setVolumeRelativeChangeFactor(v.floatValue()));
```

Comment:
We support mapping configuration to a configuration class, it would cover some or more parts of your use case. Its not needed to change, then I should have joined earlier.

@wborn do you have any advice for a next PR how to handle the configuration, would using the config class solve this (partly)?

Reply:
Configuration beans (there's still no good name for this pattern within esh), make the code much more readable. They do automatic conversions and can have default values.

Reply:
Yes using binding specific configuration objects simplifies the code. When you override `updateConfiguration(..)` instead you can easily get the configuration object and map it to a binding specific configuration:

```java
@Override 
protected void updateConfiguration(Configuration configuration) { 
    zoneConfiguration = configuration.as(YamahaZoneConfiguration.class); 
    super.updateConfiguration(configuration); 
} 
```

Calling the super is also important to keep the handler behaving like a default BaseThingHandler. I used something similar in the [PlugwiseRelayDeviceHandler](https://github.com/openhab/openhab2-addons/blob/ead0cc3953ac3c68a862eab107506f677d26980b/addons/binding/org.openhab.binding.plugwise/src/main/java/org/openhab/binding/plugwise/handler/PlugwiseRelayDeviceHandler.java#L554).

When the handler is initialized you get the configuration with:

```java
@Override
public void initialize() {
    zoneConfiguration = getConfigAs(YamahaZoneConfiguration.class);
}

```

Reply:
@wborn this is really neat, but how can I make it map a config value let say "ZONE" (or "VOLUME_DB_MIN") to the config bean property (zone, volumeDbMin)? Is there some annotation I could use to define the mapping?

Looking at ConfigMapper source from ESH, it seems expect naming in config to match bean properties.

Without this, I would have to rename the configs (breaking backwards compatibility) or set weird property names.

Reply:
I proposed annotations for the configuration parser, but they got denied. Configuration keys should be lowercase, latin letters anyway, so it might be necessary to break backwards compatibility for the purpose of clean maintainable code.

Reply:
@martinvw @davidgraeff given the limitations of configuration bean mappings (bean properties need to match settings name) and your advise to refactor this, I have introduced the breaking change. Documentation has been updated. Essentially, users will be forced to re-update their settings when migrating to this version. I hope this will not cause too much issues for the users.